Remotion/Core/Development.Analyzers.IntegrationTests/RMCORE0001_SafeContextAnalyzerTests.cs

```diff
@@ -0,0 +1,236 @@
+
+    Parallel.ForEachAsync(
+        Enumerable.Range(0, 5),
+        new ParallelOptions(), 
+        (i, token) => ValueTask.CompletedTask
+    );
+
+    Parallel.Invoke(() => Console.Write(""a""));
+  }
+}
+";
+      var diagnosticFor = Verifier.Diagnostic(Analyzer.AlternativeDescriptor)
+          .WithSpan(12, 5, 12, 77)
+          .WithMessage(
+              "'System.Threading.Tasks.Parallel.For' should not be used with SafeContext by itself, use 'provider.OpenSafeContextBoundary()' to create a safe context boundary.");
+
+      var diagnosticForEach = Verifier.Diagnostic(Analyzer.AlternativeDescriptor)
+          .WithSpan(14, 5, 14, 78)
+          .WithMessage(
+              "'System.Threading.Tasks.Parallel.ForEach' should not be used with SafeContext by itself, use 'provider.OpenSafeContextBoundary()' to create a safe context boundary.");
+
+      var diagnosticForEachAsync = Verifier.Diagnostic(Analyzer.AlternativeDescriptor)
+          .WithSpan(16, 5, 20, 6)
+          .WithMessage(
+              "'System.Threading.Tasks.Parallel.ForEachAsync' should not be used with SafeContext by itself, use 'provider.OpenSafeContextBoundary()' to create a safe context boundary.");
+
+      var diagnosticInvoke = Verifier.Diagnostic(Analyzer.AlternativeDescriptor)
+          .WithSpan(22, 5, 22, 46)
+          .WithMessage(
+              "'System.Threading.Tasks.Parallel.Invoke' should not be used with SafeContext by itself, use 'provider.OpenSafeContextBoundary()' to create a safe context boundary.");
+
+      await Verifier.VerifyAnalyzerAsync(input, true,diagnosticFor, diagnosticForEach, diagnosticForEachAsync, diagnosticInvoke);
+    }
+
+    [Test]
+    public async Task SafeContext_NewThread_Diagnostic ()
+    {
+      const string input = @"
+using System.Threading;
+using Remotion.Context;
+public class A
+{
+  public void Test ()
+  { 
+    SafeContext.Instance.SetData(""a"", 3);
+    new Thread(() => SafeContext.Instance.GetData(""a""));
+  }
+}
+";
+      var diagnosticFunc = Verifier.Diagnostic(Analyzer.AlternativeDescriptor)
+          .WithSpan(9, 5, 9, 56)
+          .WithMessage(
+              "'new System.Threading.Thread' should not be used with SafeContext, use 'Remotion.Context.SafeContext.Thread.New' instead.");
+
+      await Verifier.VerifyAnalyzerAsync(input, true,diagnosticFunc);
+    }
+
+    [Test]
+    public async Task SafeContext_NewThreadingTimer_Diagnostic ()
+    {
+      const string input = @"
+using System.Threading;
+using Remotion.Context;
+public class A
+{
+  public void Test ()
+  { 
+    SafeContext.Instance.SetData(""a"", 3);
+    var timer = new Timer(new TimerCallback(_ => { }));
+  }
+}
+";
+      var diagnosticFunc = Verifier.Diagnostic(Analyzer.AlternativeDescriptor)
+          .WithSpan(9, 17, 9, 55)
+          .WithMessage(
+              "'new System.Threading.Timer' should not be used with SafeContext by itself, use 'provider.OpenSafeContextBoundary()' to create a safe context boundary.");
```

Comment:
`SafeContext.Threading.NewTimer`

Reply:
Do not know whether this should be the alternative or the message, did not find `Safecontext.Threading.NewTimer` or anything with a timer in safecontext, really.

Reply:
You need to rebase to see the new classes regarding `ContinueWith`, and all the timers. The issue was still being merged when you started working on this issue.